apps/testapp/src/components/RpcMethods/RpcMethod.ts

```diff
@@ -0,0 +1,14 @@
+type FormattedParamsType = Record<string, unknown> | string;
+
+type ShortcutType = {
+  key: string;
+  data: Record<string, string>;
+};
+
+export type RpcMethod = {
+  connected?: boolean;
+  method: string;
+  params: Array<{ key: string; required?: boolean }>;
+  format?: (data: Record<string, string>) => FormattedParamsType[];
+  shortcuts?: ShortcutType[];
+};
```

Comment:
@fan-zhang-sv  we already discussed offline, but for the record here.
let's explore whether we can achieve this without introducing new type.

`RpcMethod` could be misleading. it seems this type is an extension of `Request` type. how about something like
```suggestion
export type RpcRequestInput = {
  method: RPCMethod; // strongly typed
};
```
`ShortcutType` might better be managed separately from the RPC method, probably a map between `method` and set of `shortcuts`? it's just rough idea.

i will also take a look at the SDK code to see if any of the internal types we have could be exposed in some way for type checking.

Reply:
yup yup,

- added separate ShortcutType
- renamed to RpcRequestInput, and have method field strongly typed
- created mapping to load shortcuts for method